hedera-node/src/main/java/com/hedera/services/txns/consensus/TopicUpdateTransitionLogic.java

```diff
@@ -55,247 +55,151 @@
-		applyNewFields(op, mutableTopic);
-		transactionContext.setStatus(SUCCESS);
-	}
-
-	private boolean wantsToMutateNonExpiryField(ConsensusUpdateTopicTransactionBody op) {
-		return op.hasMemo() ||
-				op.hasAdminKey() || op.hasSubmitKey() ||
-				op.hasAutoRenewPeriod() || op.hasAutoRenewAccount();
 	}
 
-	private boolean canApplyNewFields(ConsensusUpdateTopicTransactionBody op, MerkleTopic topic) {
-		return canApplyNewKeys(op, topic) &&
-				canApplyNewMemo(op) &&
-				canApplyNewExpiry(op, topic) &&
-				canApplyNewAutoRenewPeriod(op) &&
-				canApplyNewAutoRenewAccount(op, topic);
+	private void validateNewAdminKey(final Key newAdminKey, final TopicID topicId) {
+		String errorMessage = String.format("Update topic %s has invalid admin key specified, " +
+				"which should have been caught during signature validation", topicId);
+		ValidationUtils.validateFalse(!validator.hasGoodEncoding(newAdminKey), BAD_ENCODING, log::error, errorMessage);
 	}
 
-	private void applyNewFields(ConsensusUpdateTopicTransactionBody op, MerkleTopic topic) {
-		applyNewKeys(op, topic);
-		applyNewMemo(op, topic);
-		applyNewExpiry(op, topic);
-		applyNewAutoRenewPeriod(op, topic);
-		applyNewAutoRenewAccount(op, topic);
-	}
-
-	private boolean canApplyNewAutoRenewPeriod(ConsensusUpdateTopicTransactionBody op) {
-		if (!op.hasAutoRenewPeriod()) {
-			return true;
-		}
-		var newAutoRenewPeriod = op.getAutoRenewPeriod();
-		if (!validator.isValidAutoRenewPeriod(newAutoRenewPeriod)) {
-			transactionContext.setStatus(AUTORENEW_DURATION_NOT_IN_RANGE);
-			return false;
+	private void validateAutoRenewAllowance(final Key newAdminKey,
+											final TopicID topicId,
+											final ConsensusUpdateTopicTransactionBody op,
+											final Topic topic
+	) {
+		JKey jKey = validator.attemptDecodeOrThrow(newAdminKey, topicId);
+		if (jKey.isEmpty()) {
+			final var opRemovesAutoRenewId = op.hasAutoRenewAccount() &&
+					designatesAccountRemoval(op.getAutoRenewAccount());
+			validateFalse(topic.hasAutoRenewAccountId() && !opRemovesAutoRenewId, AUTORENEW_ACCOUNT_NOT_ALLOWED);
 		}
-		return true;
 	}
 
-	private boolean canApplyNewMemo(ConsensusUpdateTopicTransactionBody op) {
-		if (!op.hasMemo()) {
-			return true;
-		}
-		var memoValidity = validator.memoCheck(op.getMemo().getValue());
-		if (memoValidity != OK) {
-			transactionContext.setStatus(memoValidity);
-			return false;
-		}
-		return true;
+	private void validateNewSubmitKey(final ConsensusUpdateTopicTransactionBody op, final TopicID topicId) {
+		validateFalse(!validator.hasGoodEncoding(op.getSubmitKey()), BAD_ENCODING);
+		validator.attemptDecodeOrThrow(op.getSubmitKey(), topicId);
 	}
 
-	private void applyNewMemo(ConsensusUpdateTopicTransactionBody op, MerkleTopic topic) {
-		if (op.hasMemo()) {
-			topic.setMemo(op.getMemo().getValue());
-		}
+	private void validateTopicIsAuthorized(final Topic topic, final ConsensusUpdateTopicTransactionBody op) {
```

Comment:
I am not sure that `authorized` is something that makes sense to me. It sounds confusing. What does it mean for a topic to be authorised to do something?

I think that this logic is equivalent to topic's -> `affectsExpiryAtMost` and if I as a dev see `affectsExpiryAtMost` will make sense to me why we check the admin key.
What about:
```Java
validateFalse(topic.adminKey() == null && !affectsExpiryAtMost, UNAUTHORIZED)

Reply:
In the original implementation of the TopicUpdateTransitionLogic the author named this composite boolean wantsToMutateNonExpiryField. 

When refactoring - I usually do not change the names of local variables, unless they are misleading.

![image](https://user-images.githubusercontent.com/77110630/135206614-2ba230b2-f4ca-4690-90e5-30486fee4cc9.png)

Regardless of the above - the boolean is now affectsExpiryAtMost